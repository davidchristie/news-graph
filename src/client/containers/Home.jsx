import React, { PropTypes } from 'react'
import { connect } from 'react-redux'

import ArticleCards from '../components/ArticleCards'
import Jumbotron from '../components/Jumbotron'

export const Home = class Home extends React.Component {
  render () {
    const title = 'Home'
    const articles = this.props.articles
    return (
      <div>
        <Jumbotron title={title} />
        <ArticleCards articles={articles} />
      </div>
    )
  }
}

Home.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired
  }).isRequired).isRequired
}

function mapStateToProps (state) {
  return {
    articles: state.articles
  }
}

export default connect(mapStateToProps)(Home)
