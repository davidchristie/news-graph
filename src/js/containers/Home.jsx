import React, { PropTypes } from 'react'
import { connect } from 'react-redux'

import ArticleCards from '../components/ArticleCards'
import PostArticle from './PostArticle'

const Home = class Home extends React.Component {
  render() {
    const articles = this.props.articles.reverse()
    return (
      <div>
        <div className="jumbotron">
          <h1 className="display-3">Home</h1>
          <p className="lead">This is a simple hero unit, a simple jumbotron-style component for calling extra attention to featured content or information.</p>
          <hr className="my-4" />
          <p>It uses utility classes for typography and spacing to space content out within the larger container.</p>
          <PostArticle />
        </div>
        <ArticleCards articles={articles} />
      </div>
    )
  }
}

Home.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    url: PropTypes.string.isRequired
  }).isRequired).isRequired
}

function mapStateToProps(state) {
  return {
    articles: state.app.articles
  }
}

export default connect(mapStateToProps)(Home)
