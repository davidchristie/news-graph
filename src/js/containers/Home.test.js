/* eslint-env jest */

import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import Home from './Home'
import initialState from '../examples/initial-state'

const store = configureStore()(initialState)

test('Home renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Home />
    </Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
