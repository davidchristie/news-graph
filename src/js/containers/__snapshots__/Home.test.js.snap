exports[`test Home renders correctly 1`] = `
<div>
  <div>
    <div
      className="jumbotron">
      <h1
        className="display-3">
        Home
      </h1>
      <p
        className="lead">
        This is a simple hero unit, a simple jumbotron-style component for calling extra attention to featured content or information.
      </p>
      <hr
        className="my-4" />
      <p>
        It uses utility classes for typography and spacing to space content out within the larger container.
      </p>
      <form
        onSubmit={[Function]}>
        <label>
          Article:
          <input
            onChange={[Function]}
            type="url"
            value="" />
        </label>
        <input
          type="submit"
          value="POST" />
      </form>
    </div>
  </div>
  <div
    className="album text-muted">
    <div
      className="container">
      <div
        className="row">
        <div
          className="card-group">
          <div
            className="card col-md-6 col-lg-4">
            <a
              className="embedly-card"
              href="http://www.nature.com/news/ancient-retroviruses-emerged-half-a-billion-years-ago-1.21274" />
            <p>
              Some details about the article.
            </p>
            <a
              className="btn"
              onClick={[Function]}
              style={Object {}}>
              Details
            </a>
          </div>
          <div
            className="card col-md-6 col-lg-4">
            <a
              className="embedly-card"
              href="https://www.theguardian.com/world/2017/jan/11/chamber-of-rats-mexico-parliaments-name-changed-in-google-maps-prank" />
            <p>
              Some details about the article.
            </p>
            <a
              className="btn"
              onClick={[Function]}
              style={Object {}}>
              Details
            </a>
          </div>
          <div
            className="card col-md-6 col-lg-4">
            <a
              className="embedly-card"
              href="https://vimeo.com/192574852" />
            <p>
              Some details about the article.
            </p>
            <a
              className="btn"
              onClick={[Function]}
              style={Object {}}>
              Details
            </a>
          </div>
          <div
            className="card col-md-6 col-lg-4">
            <a
              className="embedly-card"
              href="http://www.aljazeera.com/indepth/features/2017/01/india-love-hand-books-170108073151216.html" />
            <p>
              Some details about the article.
            </p>
            <a
              className="btn"
              onClick={[Function]}
              style={Object {}}>
              Details
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
